lib/Zef/Repository/Ecosystems.rakumod

```diff
@@ -154,9 +158,15 @@ class Zef::Repository::Ecosystems does PackageRepository {
             my $wanted-short-name   := $wanted-spec.name;
             my $dists-to-search     := $strict ?? (%!short-name-lookup{$wanted-short-name} // Nil).grep(*.so) !! %!short-name-lookup{%!short-name-lookup.keys.grep(*.contains($wanted-short-name))}.map(*.Slip).grep(*.so);
             my $matching-candidates := $dists-to-search.grep(*.contains-spec($wanted-spec, :$strict)).map({
+                my $uri;
+                if $_.meta<path> && $.uses-path {
+                    $uri = $_.meta<path>;
+                    $uri ~~ s/^repo\///;
+                    $uri = $.mirrors.first ~ $uri;
+                }
                 Candidate.new(
                     dist => $_,
-                    uri  => ($_.source-url || $_.meta<support><source>),
+                    uri  => ($uri || $_.source-url || try $_.hash<support><source>),
```

Comment:
https://github.com/ugexe/zef/blob/1cbd322f6da5e1b67245dae98751463919a0229c/lib/Zef/Repository/Ecosystems.rakumod#L159 this is what it should be

Reply:
That dies on the ci about odd number of elements in the hash

Reply:
That is what it currently is though, which does work. Something in this PR appears to break that.
https://github.com/ugexe/zef/blob/1cbd322f6da5e1b67245dae98751463919a0229c/lib/Zef/Repository/Ecosystems.rakumod#L159